src/site/content/en/blog/aria-poison-or-antidote/index.md

```diff
@@ -0,0 +1,283 @@
+item,  the screen reader now knows where the user has moved, but nothing else about the object. What
+is this div thing anyway? That's where the role attribute comes in. We use `role="menubar"` on the
+containing element for the entire thing, then we use `role="menu"` on groups of items, and
+`role="menuitem"` on … drumroll … the individual menu items.
+
+And what if the menuitem can lead to a child menu? The user needs to know that right? For a sighted
+user, there might be a little picture of a triangle at the end of the menu, but the screen reader
+doesn't know how to automatically read images, at least at this point. We can add
+`aria-expanded="false"` on each expandable menuitem to indicate that 1) there is something that can
+be expanded, and 2) it currently is not expanded. As an added touch the author should put
+`role="none"` on the img triangle to indicate it's for prettification purposes only. This prevents
+the screen reader from saying anything annoying about the image that would be redundant at best.
+
+## Dealing with bugs
+
+### Keyboard bugs (HTML!)
+
+Although keyboard access is a part of core HTML, authors mess it up all the time, either because
+they don't use keyboard navigation all that much, or because there is much nuance to get right. 
+
+Examples of bugs:
+
++   A checkbox uses spacebar to toggle, but the author forgot to call `preventDefault()`. Now
+    the spacebar will both toggle the checkbox and page down, which is the default browser behavior
+    for spacebar.
++   An ARIA modal dialog wants to trap tab navigation inside of it, and the author forgets to
+    specifically allow <kbd>Control</kbd>+<kbd>Tab</kbd> through to the browser. Now,
+    <kbd>Control</kbd>+<kbd>Tab</kbd> just navigates within their dialog, and doesn't switch tabs
+    in the browser as it should. Ugh.
++   An author creates a selection list, and implements up/down, but does not implement
+    home/end/pageup/pagedown or first letter navigation. Authors should follow known patterns.
+    Check out the [Resources](#resources) section for more info.
+
+For pure keyboard access issues, it's useful to also try without a screen reader, or with virtual
+browser mode off. Screen readers are not generally necessary to exhibit a keyboard bug, and keyboard
+access is actually implemented with HTML, not ARIA. After all, ARIA doesn't affect basic stuff like
+the keyboard or mouse behavior, it only lies to the screen reader about what's in the web page,
+what's currently focused, etc. 
+
+Keyboard bugs are almost always a bug in the web content, specifically in their HTML and JavaScript,
+not in ARIA.
+
+### ARIA bugs… why are there so many?
+
+There are many, many places where authors can get ARIA wrong, and each will lead to either complete
+breakage or subtle differences. The subtle ones are probably worse, because the author won't catch
+most of them on their own.
```

Comment:
```suggestion
most of them before publishing.
```

Reply:
I'm not in love with my wording. I found "on their own" to be vague and think it needs to be changed. I think you're saying the author will not catch the errors him or herself. Leaving it off will also solve the problem.